### PR TITLE
No masking without Arguments

### PR DESCRIPTION
**Pull request by anshsrtv**
To get exact account numbers, as present in the bank feeds. Adding the condition that, if the masking arguments aren't passed to the constructor (i.e. if `account_number_mask_begin` and `account_number_mask_end` are `None`), the card numbers aren't masked by the parsers.

### CODE REVIEW

File: card_data_parsers/parsers/amex_parser.py [hunk 1]
```diff
@@ -147,8 +147,12 @@ def __extract_transactions(txn_lines, account_number_mask_begin, account_number_
                 raise ParserError(f'Currency is missing')
 
             # Masking the card number
-            txn.account_number = mask_card_number(txn.account_number, account_number_mask_begin,
-                                                    account_number_mask_end)
+            if account_number_mask_begin!=None and account_number_mask_end!=None:
```
Comment (Code0987) [inline card_data_parsers/parsers/amex_parser.py:150]: Let's change `!= None` to `is not None`. Although it works here but `None` is not a number so we should not be doing equality check of it.

Ref: https://stackoverflow.com/a/2209781/496841

**Review by Code0987 — CHANGES_REQUESTED** [review]
Pls add test cases.

Idea:
- Check `card_data_parsers/amex/p1/`
- Make a duplicate folder of it and name it p2, update the args.json with null on `account_number_mask_begin` and `account_number_mask_end`
- Update expected.json as required based on data file

**Review by Code0987 — APPROVED** [review]
Lgtm.

Pls add version number changes before merge - example: https://github.com/fylein/fyle-bank-feed-parser-sdk-py/commit/9bbcfaa9653bbd0923f2d00ee73a35853c03b471

**Review by Code0987 — APPROVED** [review]

File: card_data_parsers/parsers/vcf_parser.py [hunk 2]
```diff
@@ -274,7 +279,7 @@ def __cleanup_fields(line) -> str:
         return line
 
     @staticmethod
-    def parse(file_obj, account_number_mask_begin, account_number_mask_end, default_values={}, mandatory_fields=[]) -> List[VCFTransaction]:
+    def parse(file_obj, account_number_mask_begin=None, account_number_mask_end=None, default_values={}, mandatory_fields=[]) -> List[VCFTransaction]:
```
Comment (modernwarfareuplink) [inline card_data_parsers/parsers/vcf_parser.py:282]: @anshsrtv can you create a clickup to remove using list as default parameters pls.

**Review by modernwarfareuplink — APPROVED** [review]

**Review by Kulsekar4 — APPROVED** [review]

**Review by gokulk16 — APPROVED** [review]

**Pull request merged**